great_expectations/datasource/data_connector/inferred_asset_s3_data_connector.py

```diff
@@ -93,6 +93,22 @@ def __init__(
                 "Unable to load boto3 (it is required for InferredAssetS3DataConnector)."
             )
 
+    @staticmethod
+    def sanitize_prefix(text: str) -> str:
```

Comment:
Could we change the name of this to `sanitize_prefix_for_s3` unless this will be used for anything else?

Additionally, would you please add this to the `ConfiguredAssetS3DataConnector`?